src/OpenTelemetry/DiagnosticSourceInstrumentation/DiagnosticSourceListener.cs

```diff
@@ -41,7 +41,11 @@ public void OnNext(KeyValuePair<string, object> value)
         {
             if (!this.handler.SupportsNullActivity && Activity.Current == null)
             {
-                InstrumentationEventSource.Log.NullActivity(value.Key);
+                if (!Sdk.SuppressInstrumentation)
```

Comment:
should we add this check inside the instrumentationEventSource? because we can forget if someone adds a new Log, no?

Reply:
Good question. If you look at this method there are 4 or 5 `SuppressInstrumentation` checks inside it. Smells like it could use a bit of cleanup. /cc @alanwest